### PR TITLE
Add a test for OpenX JSON SerDe

### PR DESCRIPTION
**Pull request by MiguelWeezardo**

### CODE REVIEW

File: testing/trino-product-tests/src/main/java/io/trino/tests/hive/TestAllDatatypesFromHiveConnector.java [hunk 1]
```diff
@@ -152,6 +163,68 @@ public void testSelectAllDatatypesTextFile()
...
+                        new BigDecimal("346"),
+                        new BigDecimal("345.67800"),
+                        new BigDecimal("12345678901234567890.123456789012345678"),
+                        Timestamp.valueOf(LocalDateTime.of(2015, 5, 10, 12, 15, 35, 123_000_000)),
+                        Date.valueOf("2015-05-10"),
+                        "ala ma kota",
+                        "ala ma kot",
+                        "ala ma    ",
+                        true,
+                        "kot binarny".getBytes(UTF_8)));
```
Comment (wendigo) [inline testing/trino-product-tests/src/main/java/io/trino/tests/hive/TestAllDatatypesFromHiveConnector.java:225]: :D

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/SinglenodeHdp3.java [hunk 2]
```diff
@@ -58,6 +60,21 @@ public void extendEnvironment(Environment.Builder builder)
             dockerContainer.withCopyFileToContainer(
                     forHostPath(dockerFiles.getDockerFilesHostPath("conf/environment/singlenode-hdp3/apply-hdp3-config.sh")),
                     CONTAINER_HADOOP_INIT_D + "apply-hdp3-config.sh");
+            dockerContainer.withCopyFileToContainer(
+                    forHostPath(dockerFiles.getDockerFilesHostPath("conf/environment/singlenode-hdp3/json-serde-1.3.9-SNAPSHOT-jar-with-dependencies.jar")),
+                    "/docker/openx-serde.jar");
```
Comment (sopel39) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/SinglenodeHdp3.java:65]: would it be possible to get openx as maven dependency, e.g:
```
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-dependency-plugin</artifactId>
                <executions>
                    <execution>
                        <id>copy</id>
                        <phase>package</phase>
                        <goals>
                            <goal>copy</goal>
                        </goals>
                        <configuration>
                            <skip>false</skip>
                            <artifactItems>
                                <artifactItem>
                                    <groupId>io.trino</groupId>
                                    <artifactId>trino-cli</artifactId>
                                    <version>${dep.trino.version}</version>
                                    <classifier>executable</classifier>
                                    <type>jar</type>
                                    <outputDirectory>${project.build.directory}</outputDirectory>
                                </artifactItem>
```
in product tests launcher xml

File: testing/trino-product-tests/src/main/java/io/trino/tests/hive/TestAllDatatypesFromHiveConnector.java [hunk 3]
```diff
@@ -152,6 +163,68 @@ public void testSelectAllDatatypesTextFile()
...
+                DECIMAL,
+                DECIMAL,
+                TIMESTAMP,
+                DATE,
+                VARCHAR,
+                VARCHAR,
+                CHAR,
+                BOOLEAN,
+                VARBINARY);
+        assertThat(queryResult).containsOnly(
```
Comment (sopel39) [inline testing/trino-product-tests/src/main/java/io/trino/tests/hive/TestAllDatatypesFromHiveConnector.java:208]: does it test edge cases that are specifically handled by openx?

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/SinglenodeHdp3.java (hunk 2 above)
Comment (MiguelWeezardo) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/SinglenodeHdp3.java:65]: Sure, I can do that, but what is the intention of adding it as a product tests launcher dependency? Would this impact the classpath inside Presto or Hadoop dockers? The test launcher doesn't read the data with the serde itself.

File: testing/trino-product-tests/src/main/java/io/trino/tests/hive/TestAllDatatypesFromHiveConnector.java (hunk 3 above)
Comment (MiguelWeezardo) [reply testing/trino-product-tests/src/main/java/io/trino/tests/hive/TestAllDatatypesFromHiveConnector.java:208]: I think I still need to add a couple test cases for this serde. It seems I left some features untested.

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/SinglenodeHdp3.java (hunk 2 above)
Comment (sopel39) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/SinglenodeHdp3.java:65]: The intention is not to store binary files with repo and rely on dependencies instead.

Comment (MiguelWeezardo) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/SinglenodeHdp3.java:65]: Oh, sure, I will have to upload this to a Maven repository somewhere. No problem.

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/SinglenodeHdp3.java [hunk 4]
```diff
@@ -58,6 +60,21 @@ public void extendEnvironment(Environment.Builder builder)
             dockerContainer.withCopyFileToContainer(
                     forHostPath(dockerFiles.getDockerFilesHostPath("conf/environment/singlenode-hdp3/apply-hdp3-config.sh")),
                     CONTAINER_HADOOP_INIT_D + "apply-hdp3-config.sh");
+            dockerContainer.withCopyFileToContainer(
+                    forHostPath(dockerFiles.getDockerFilesHostPath("conf/environment/singlenode-hdp3/json-serde-1.3.9-SNAPSHOT-jar-with-dependencies.jar")),
```
Comment (electrum) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/SinglenodeHdp3.java:64]: We need to use a non-snapshot version before merging

**Pull request closed without merging**